### ISSUE TITLE
Homescreen favicon flicker when dismissing tab tray

### ISSUE THREAD

**Issue by nbhasin2** [mozilla-mobile/firefox-ios#14175]
## Steps to reproduce
- Open Firefox and load a few tabs
- Make sure your homepage is set to Firefox home
- Have a few items in jump back in so that section is shown
- Tap on show all button

### Expected behavior

- When tab tray opens and dismiss via done the home page refreshes without any flicker

### Actual behaviour

- When tab tray opens and dismiss via done the home page refreshes causing favicons to flicker

### Device & build information
* Device: ? iPhone 14 pro
* Build version: v113 (30010)
* First seen version: ?
### Notes
Attachments: 
N/A

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-6299)

**Comment by nbhasin2** [comment]
I am opening the ticket to further shine light on the flicker and this also happens when backgrounding the app.

I have seen this before so please mark it a dupe if other issue exists somewhere in jira / github.

**Comment by data-sync-user** [comment]
➤ Razvan Litianu commented:

Another ticket that might be related https://mozilla-hub.atlassian.net/browse/FXIOS-5974 ( https://mozilla-hub.atlassian.net/browse/FXIOS-5974|smart-link ) .

**Comment by data-sync-user** [comment]
➤ Razvan Litianu commented:

Nishant Bhasin Daniela Arcese 

Upon my investigation, it seems that every time the user comes to the home screen the ‘collectionView` is reloading the data completely. That causes for a brief moment to set the image to nil because of cell reusability, and then the image is loaded from cache and that causes to flicker. 

One immediate action that we can do it to introduce a fade animation to make it easier for the eyes. But I think long term we should consider updating it with diffable data source and reduce the updates.

I’m attaching a recording to see how it looks. Feel free to tag someone from UX that can add some input.

**Comment by data-sync-user** [comment]
➤ Razvan Litianu commented:

Empty favicons for a split second after dismiss with no animation.

!deque-nil-image-no-animation.mov|width=886,height=1920!

Empty favicons for a split second after dismiss followed by a fade animation.

!deque-nil-image-fade-animation.mov|width=886,height=1920!

**Comment by data-sync-user** [comment]
➤ Daniela Arcese commented:

Michael Verdi what is your opinion?

**Comment by data-sync-user** [comment]
➤ Michael Verdi commented:

This is an issue that we’ve prioritized for the fundamentals work but it’s been blocked by a refactor. This happens not just when switching from the tab tray but often anytime you access the home page—startup/foregrounding, hitting the home button, hitting the back button. It also just happens if you just leave the homepage open and don’t interact with your phone for a few moments. Everything will just flickers randomly. It happens so regularly and often that I don’t think a fade animation will help much. Things will just be fading in and out constantly for no discernible reason.
